akka-stream/src/main/scala/akka/stream/impl/streamref/SinkRefImpl.scala

```diff
@@ -91,12 +91,12 @@ private[stream] final class SinkRefStageImpl[In] private[akka] (
 
         partnerRef match {
           case OptionVal.Some(ref) ⇒
-            ref ! StreamRefsProtocol.OnSubscribeHandshake(self.ref)
             tryPull()
-          case _ ⇒ // nothing to do
+          case _ ⇒
+            // only schedule timeout timer if partnerRef has not been resolved yet (i.e. if this instance of the Actor
+            // has not been provided with a valid initialPartnerRef)
+            scheduleOnce(SubscriptionTimeoutTimerKey, subscriptionTimeout.timeout)
```

Comment:
seems this can be simplified, since if initialref is some, thent his will also be some;

Reply:
In general the whole preStart can be simplified a bit to separate the two cases (being origin, or being the remote counterpart). But I'm not sure I understand which simplification is possible in this local context

Reply:
Fold it with the isDefined above this